src/stitcher/image_loader.rs

```diff
@@ -31,15 +70,16 @@ pub fn find_images(directory_path: &str) -> anyhow::Result<Vec<PathBuf>> {
             _ => false,
         })
         .collect();
+
+    // if no images were found
     if images.is_empty() {
         return Err(ImageLoaderError::NoImagesInDirectory.into());
     }
-    images.sort_by(|a, b| {
-        natord::compare(
-            a.file_name().unwrap().to_str().unwrap(),
-            b.file_name().unwrap().to_str().unwrap(),
-        )
-    });
+
+    // sort images by natural order
+    images.sort_by(|a, b| natord::compare(&a.display().to_string(), &b.display().to_string()));
```

Comment:
Favor `.display().to_string()` over `.file_name().unwrap().to_str().unwrap()`. If there are non-textual bytes in the file name, then we should just try to gloss over it as best as possible instead of crashing the program.